cursive-core/src/logger.rs

```diff
@@ -1,15 +1,71 @@
 //! Logging utilities.
 
 use lazy_static::lazy_static;
+use std::cmp::Ord;
 use std::collections::VecDeque;
-use std::sync::Mutex;
+use std::str::FromStr;
+use std::sync::{Mutex, RwLock};
 
 /// Saves all log records in a global deque.
 ///
 /// Uses a `DebugView` to access it.
+///
+/// # Examples
+///
+/// Set log levels from env vars
+///
+/// ```
+/// # use cursive_core::*;
+/// logger::set_filter_levels_with_env();
+/// logger::init();
+/// ```
+///
+/// Set log levels explicitly.
+///
+/// ```
+/// # use cursive_core::*;
+/// # use log::LevelFilter;
+/// logger::set_int_filter_level(LevelFilter::Warn);
+/// logger::set_ext_filter_level(LevelFilter::Debug);
+/// logger::init();
+/// ```
+
 pub struct CursiveLogger;
 
-static LOGGER: CursiveLogger = CursiveLogger;
+lazy_static! {
+    // Log filter level for log messages from within cursive
+    static ref INT_FILTER_LEVEL: RwLock<log::LevelFilter> = RwLock::new(log::LevelFilter::Trace);
+    // Log filter level for log messages from sources outside of cursive
+    static ref EXT_FILTER_LEVEL: RwLock<log::LevelFilter> = RwLock::new(log::LevelFilter::Trace);
+}
+
+/// Sets the internal log filter level.
+pub fn set_int_filter_level(level: log::LevelFilter) {
```

Comment:
Nit: could we call it `set_internal_filter_level`, to prevent any ambiguity around `int`?

Let's also extend `set_external_filter_level` for consistency.

I don't care as much about the lazy static global as it's not public.